### PR TITLE
Tag, you're it!

### PR DESCRIPTION
**Pull request by ch3pjw**
See second commit for the meat.

### CODE REVIEW

File: src/Clapi/Types/Tree.hs [hunk 1]
```diff
@@ -53,6 +53,8 @@ data TreeType
   | TtFloat (Bounds Float)
   | TtDouble (Bounds Double)
   | TtString Text
+  -- FIXME: kinda want this to be TtRef (Tagged Definition TypeName) but that
```
Comment (foolswood) [inline src/Clapi/Types/Tree.hs]: This cycle is annoying, I think I got around it by defining something more abstract (I think it was the kind actually) that was basically an enumeration of the different types of type that we use, and a type family to line it up with definition later. See the branch of despair for the good and bad of this I guess.

File: src/Clapi/Valuespace.hs [hunk 2]
```diff
@@ -349,15 +366,28 @@ opsTouched cops dd = fmap (const Nothing) cops <> fmap classifyDc (alToMap dd)
     classifyDc (ConstChange {}) = Nothing
     classifyDc (TimeChange m) = Just $ Map.keysSet m
 
+
+updateNsDefs
```
Comment (foolswood) [inline src/Clapi/Valuespace.hs]: :+1:

File: test/NamespaceTrackerSpec.hs [hunk 3]
```diff
@@ -141,7 +143,7 @@ spec = do
                 ]
               })
             sendRev (i, Ocd $ ocdEmpty
-              { ocdDefinitions = Map.singleton helloTn helloDef1
+              { ocdDefinitions = Map.singleton (Tagged helloTn) helloDef1
```
Comment (foolswood) [inline test/NamespaceTrackerSpec.hs]: Why not define `helloTn` as being a tagged doodad of the right type instead of tagging at each use site.

**Comment by ch3pjw** [comment]
Instead of worrying about absolute and relative paths, we're going to try namespacing client bundles too in a separate PR. The tidies from this have been included in #111 so we don't lose them.

**Pull request closed without merging**